training/src/main/java/wolox/training/controllers/UserController.java

```diff
@@ -79,10 +81,35 @@ public void delete(@PathVariable Long id) {
     })
     public Users create(
         @ApiParam(value = "user DTO in body to create it") @RequestBody Users user) {
-        return userRepository.save(user);
+
+        Users newUser = new Users();
+
+        newUser.setName(user.getName());
+        newUser.setUsername(user.getUsername());
+        newUser.setPassword(passwordEncoder.encode(user.getPassword()));
+        newUser.setBirthdate(user.getBirthdate());
+
+        return userRepository.save(newUser);
 
     }
 
+    @PutMapping("/users/{id}/password/{password}")
+    @ApiOperation(value = "Giving an id, update password of user", response = Users.class)
+    @ApiResponses(value = {
+        @ApiResponse(code = 201, message = "Succesfully update password user"),
+        @ApiResponse(code = 404, message = "User not found")
+    })
+    public Users updatePasswordUser(
```

Comment:
The proper way of making a password update is to ask for the old and new password and check that the old password matches with the one on the DB. If you want, you could also ask for _new password confirmation_ and that would be a "muy bien 10 felicitado".